src/Microsoft.Extensions.Caching.Memory/MemoryCacheOptions.cs

```diff
@@ -11,6 +11,9 @@ public class MemoryCacheOptions : IOptions<MemoryCacheOptions>
     {
         public ISystemClock Clock { get; set; }
 
+        [Obsolete("This is obsolete and will be removed in a future version.")]
+        public bool CompactOnMemoryPressure { get; set; } = true;
```

Comment:
Depends on how much of this we want to keep the same as before. Doesn't really matter either way.